File: server/js/src/parser.js
const DIRECTIVE_PATTERN = /^\s*(?:指令|AI)[：:]([^;]+);([^,]+)(?:,(.+))?\s*$/;
const PREFIX_PATTERN = /^(指令|AI)[：:]/;
const MAX_DIRECTIVE_LENGTH = 512;
const MAX_PARAMS = 10;

export class DirectiveParseError extends Error {
  constructor(message, code = 'INVALID_DIRECTIVE_FORMAT') {
    super(message);
    this.code = code;
  }
}

export function normalizeDirectiveKey(key) {
  for (const prefix of ['指令:', 'AI:', '指令：', 'AI：']) {
    if (key.startsWith(prefix)) {
      return key.slice(prefix.length);
    }
  }
  return key;
}

export function parseDirective(prompt) {
  if (!prompt || !prompt.trim()) {
    throw new DirectiveParseError('prompt is required');
  }

  const cleaned = prompt.trim();

  if (cleaned.length > MAX_DIRECTIVE_LENGTH) {
    throw new DirectiveParseError(
      `directive exceeds max length (${MAX_DIRECTIVE_LENGTH})`
    );
  }

  const match = DIRECTIVE_PATTERN.exec(cleaned);
  if (!match) {
    throw new DirectiveParseError(`invalid directive format: ${cleaned}`);
  }

  const domain = match[1].trim();
  const action = match[2].trim();
  const rawParams = match[3];

  const params = [];
  if (rawParams) {
    for (const p of rawParams.split(',')) {
      const trimmed = p.trim();
      if (trimmed) params.push(trimmed);
    }
  }

  if (params.length > MAX_PARAMS) {
    throw new DirectiveParseError(
      `too many parameters (${params.length}), max ${MAX_PARAMS}`
    );
  }

  for (const p of params) {
    if (/[,;\n\r]/.test(p)) {
      throw new DirectiveParseError(`parameter contains forbidden characters: ${p}`);
    }
  }

  if (!domain) throw new DirectiveParseError('domain is empty');
  if (!action) throw new DirectiveParseError('action is empty');

  const prefixMatch = PREFIX_PATTERN.exec(cleaned);
  const prefix = prefixMatch ? prefixMatch[1] : '指令';

  return {
    domain,
    action,
    params,
    raw: cleaned,
    directiveKey: `${prefix}:${domain};${action}`,
  };
}

File: deploy/A5-endpoint/python/core/parser.py
import re
from dataclasses import dataclass

# 双前缀协议：中文 `指令:` 和英文 `AI:` 同等效力
# 兼容 v1.0 `指令:` 格式，同时支持 v1.1+ `AI:` 国际化前缀
DIRECTIVE_PATTERN = re.compile(
    r"^(?:指令|AI)[：:]([^;]+);([^,]+)(?:,(.+))?$"
)

# Extract prefix for directive_key reconstruction
_PREFIX_PATTERN = re.compile(r"^(指令|AI)[：:]")

MAX_DIRECTIVE_LENGTH = 512
MAX_PARAMS = 10


@dataclass
class ParsedDirective:
    domain: str
    action: str
    params: list[str]
    raw: str

    @property
    def directive_key(self) -> str:
        """返回带原始前缀的指令键，用于 schema 匹配"""
        m = _PREFIX_PATTERN.match(self.raw)
        prefix = m.group(1) if m else "指令"
        return f"{prefix}:{self.domain};{self.action}"


class DirectiveParseError(Exception):
    def __init__(self, message: str, code: str = "INVALID_DIRECTIVE_FORMAT"):
        self.message = message
        self.code = code
        super().__init__(message)


def parse_directive(prompt: str | None) -> ParsedDirective:
    if not prompt or not prompt.strip():
        raise DirectiveParseError("prompt is required")

    prompt = prompt.strip()

    if len(prompt) > MAX_DIRECTIVE_LENGTH:
        raise DirectiveParseError(
            f"directive exceeds max length ({MAX_DIRECTIVE_LENGTH})"
        )

    match = DIRECTIVE_PATTERN.match(prompt)
    if not match:
        raise DirectiveParseError(f"invalid directive format: {prompt}")

    domain = match.group(1).strip()
    action = match.group(2).strip()
    raw_params = match.group(3)

    params = []
    if raw_params:
        for p in raw_params.split(","):
            p = p.strip()
            if p:
                params.append(p)

    if len(params) > MAX_PARAMS:
        raise DirectiveParseError(
            f"too many parameters ({len(params)}), max {MAX_PARAMS}"
        )

    for p in params:
        if any(c in p for c in (",", ";", "\n", "\r")):
            raise DirectiveParseError(f"parameter contains forbidden characters: {p}")

    if not domain:
        raise DirectiveParseError("domain is empty")
    if not action:
        raise DirectiveParseError("action is empty")

    return ParsedDirective(
        domain=domain,
        action=action,
        params=params,
        raw=prompt,
    )
